src/services/inlayHints.ts

```diff
@@ -473,6 +472,11 @@ export function provideInlayHints(context: InlayHintsContext): InlayHint[] {
                 return;
             }
 
+            if (isLiteralExpression(node)) {
+                parts.push({ text: isStringLiteral(node) ? `"${node.text}"` : node.text });

```

Comment:
This has to be escaped technically. Can you add a test for the string literals `'"'` and `"\\"`?

Reply:
Would it be an overkill to use the quote kind from `UserPreferences` here?

Reply:
That's probably the ideal thing to do.

Sorry for not making this an easy fix @a-tarasyuk. 😄

Reply:
@MariaSolOs Is this how you wanted to use `quotePreference` https://github.com/microsoft/TypeScript/pull/55947/commits/9700893341028bdc53989df7c6623b3c0b4af10c?

Reply:
@a-tarasyuk It is. I just have a couple of concerns and I want to check in with @DanielRosenwasser first. Sorry for all the trouble!